qemu/tests/qemu_guest_agent.py

```diff
@@ -343,11 +343,35 @@ def setup(self, test, params, env):
                     test.fail("Failed to get qga, details: %s" % output)
                 else:
                     self.qga_pkg_path = params["qga_rpm_path"]
-            if self._check_ga_pkg(session, params.get("gagent_pkg_check_cmd")):
-                LOG_JOB.info("qemu-ga is already installed.")
+            if self.params["gagent_src_type"] != "url":
+                if self._check_ga_pkg(session, params.get("gagent_pkg_check_cmd")):
+                    LOG_JOB.info("qemu-ga is already installed.")
+                else:
+                    LOG_JOB.info("qemu-ga is not installed or need to update.")
+                    self.gagent_install(session, self.vm)
             else:
-                LOG_JOB.info("qemu-ga is not installed or need to update.")
-                self.gagent_install(session, self.vm)
+                error_context.context("Download qemu-guest-agent package from website "
+                                      "and copy it to guest.", logging.info)
+                gagentrpm_download_url = self.params["gagent_download_url"]
+                gagentrpm_guest_dir = self.params["gagentrpm_guest_dir"]
+                gagentrpm_tem_path = self.params["gagentrpm_tem_path"]
+                rpm_name = re.search(r'([^/]+\.rpm)$', gagentrpm_download_url).group(1)
+                process.system(f'wget -qP {gagentrpm_tem_path} {gagentrpm_download_url}')
```

Comment:
Hi @6-dehan I guess we can download the gagent rpm in VM directly, it will simply our code and test steps.What do you think?

Reply:
Hi @leidwang , thanks for the reviewing, but the previous logic to download mingw-qemu-ga was the same logic. For qga I want to keep them same. and sometimes download inside VM would failed since the network issue. so keep it it's fine .